File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/TerastallizeModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.PokeType;
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class TerastallizeModifierItem extends PokemonHeldItemModifierItem implements GeneratedPersistentModifierType {

    public static final String TARGET = "TerastallizeModifierType";

    private PokeType teraType;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/TerastallizeModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.PokeType
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType

class TerastallizeModifierItem : PokemonHeldItemModifierItem(), GeneratedPersistentModifierType {

    companion object {
        const val TARGET = "TerastallizeModifierType"
    }

    var teraType: PokeType? = null
}